.github/workflows/run_pipeline.yaml

```diff
@@ -0,0 +1,81 @@
+# Purpose:  Run the CI/CD pipeline that tests, packages, and publishes ScubaGear.
+# Note:  This pipeline is a work in progress.  At the moment, it is only doing linting, syntax checking, and unit testing.
+
+name: CI/CD Pipeline
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      # When set to true, it will run every step in the pipeline, regardless of
+      # what files have changed.
+      doEverything:
+        description: "Run every workflow in the pipeline."
+        required: false
+        type: boolean
+        default: true
+
+jobs:
+  test-files:
+    name: Test for Changes
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Check for Changes
+        uses: dorny/paths-filter@v3
+        id: file-changes
+        with:
+          base: ${{ github.ref }}
+          filters: |
+            yaml-files:
+              - added|modified: "**.yml"
+              - added|modified: "**.yaml"
+            powershell-files:
+              - added|modified: "**.ps1"
+              - added|modified: "**.psm1"
+              - added|modified: "**.psd1"
+              - added|modified: "**.pssc"
+              - added|modified: "**.psrc"
+              - added|modified: "**.ps1xml"
+              - added|modified: "**.cdxml"
+            markdown-files:
+              - added|modified: "PowerShell/ScubaGear/baselines/*.md"
+            rego-files:
+              - added|modified: "**.rego"
+    outputs:
+      yaml-changes: ${{ steps.file-changes.outputs.yaml-files || inputs.doEverything }}
+      powershell-changes: ${{ steps.file-changes.outputs.powershell-files || inputs.doEverything }}
+      markdown-changes: ${{ steps.file-changes.outputs.markdown-files || inputs.doEverything }}
+      rego-changes: ${{ steps.file-changes.outputs.rego-files || inputs.doEverything }}
+  lint-yaml:
+    name: Lint1
+    needs:
+      - test-files
+    if: needs.test-files.outputs.yaml-changes == 'true'
+    uses: cisagov/ScubaGear/.github/workflows/run_linter_yaml.yaml@main
+  lint-powershell:
+    name: Lint2
```

Comment:
The workflow already does what you want, though it's not obvious from the code.  The calling workflow pulls the name of the reusable workflow into the little flowchart diagram, so you know exactly which linting is happening.
![image](https://github.com/cisagov/ScubaGear/assets/52328727/e6cdeafa-8af9-426e-81de-28aaefef906c)
There's an outstanding issue to clean up the names of our workflows (859), so that will make this even more obvious.

Reply:
I removed the 1 and 2.  See above for the rest.